packages/medusa-plugin-sendgrid/src/services/sendgrid.js

```diff
@@ -184,7 +184,15 @@ class SendGridService extends NotificationService {
   }
 
   getTemplateId(event) {
-    switch (event) {
+    const templates = Object.keys(this.options_ ?? {})
+    const key = templates.find(
+      (template) =>
+        event.toLowerCase().toLowerCase().replace(".", "_") === template ||
+        `${event.toLowerCase().toLowerCase().replace(".", "_")}_template` ==
+          template
+    )
```

Comment:
comment: Can we try to make this more clean? E.g.:
```suggestion
    const templates = Object.keys(this.options_ ?? {})
    
    const key = templates.find(
      (template) =>
       {
         const formattedEvent = event.toLowerCase().replace(".", "_")
         return formattedEvent === template || `${formattedEvent}_template` === template
       }
    )
```

Reply:
agreed